packages/melos/lib/src/common/utils.dart

```diff
@@ -205,6 +209,77 @@ class PromptException extends MelosException {
   String toString() => 'Was unable to prompt for input:\n$message';
 }
 
+class PersistentShell {
+  PersistentShell({
+    required MelosLogger logger,
+    String? workingDirectory,
+  }) {
+    _logger = logger;
+    _workingDirectory = workingDirectory;
+  }
+
+  final _isWindows = currentPlatform.isWindows;
+  late final MelosLogger _logger;
+  late final String? _workingDirectory;
```

Comment:
These don't have to be private, they can be set directly in the constructor like this:
```dart
  PersistentShell({
    required MelosLogger logger,
    String? workingDirectory,
  }) : _logger = logger,  _workingDirectory = workingDirectory;
  ```
  
  But I don't think there is any reason to keep them private, so you can simplify it even further:
  ```dart
  PersistentShell({
    required this.logger,
    this.workingDirectory,
  });
  ```